src/passes/builtinHandler/require.ts

```diff
@@ -50,33 +48,29 @@ export class Require extends ASTMapper {
     }
 
     if (expression.vIdentifier === 'require' || expression.vIdentifier === 'assert') {
-      const requireMessage =
-        expression.vArguments[1] instanceof Literal ? expression.vArguments[1].value : null;
-
       return new ExpressionStatement(
         ast.reserveId(),
         expression.src,
-        new CairoAssert(
+        new FunctionCall(
           ast.reserveId(),
           expression.src,
-          expression.vArguments[0],
-          requireMessage,
-          expression.raw,
+          expression.typeString,
+          expression.kind,
+          expression.vExpression,
+          expression.vArguments,
         ),
       );
     } else if (expression.vIdentifier === 'revert') {
-      const revertMessage =
-        expression.vArguments[0] instanceof Literal ? expression.vArguments[0].value : null;
-
       return new ExpressionStatement(
         ast.reserveId(),
         expression.src,
-        new CairoAssert(
+        new FunctionCall(
           ast.reserveId(),
           expression.src,
+          expression.typeString,
+          expression.kind,
           createBoolLiteral(false, ast),
```

Comment:
This boolean literal can't be the expression of the Function Call node, this argument is supposed to be the expression that defines the callee, in this case it should be an Identifier node "revert". You can just pass `expression.vExpression` for that as you did in the previous case.

You still need to add `false` as the first parameter of this function though, so it gets printed correctly when it gets to the writer. For that you will need to modify the list of arguments you pass to create this FunctionCall node. Instead of passing `expression.vArguments`, it could be:
 `[createBoolLiteral(false, ast), ...expression.vArguments]`

Reply:
You should also add some examples to compilation tests to check the transpilation was correct.